pkg/repositories/transformers/description_entity.go

```diff
@@ -0,0 +1,94 @@
+package transformers
+
+import (
+	"context"
+
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+
+	"github.com/flyteorg/flyteadmin/pkg/errors"
+	"github.com/flyteorg/flyteadmin/pkg/repositories/models"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flytestdlib/logger"
+	"github.com/golang/protobuf/proto"
+	"google.golang.org/grpc/codes"
+)
+
+// CreateDescriptionEntityModel Transforms a TaskCreateRequest to a Description entity model
+func CreateDescriptionEntityModel(
+	descriptionEntity *admin.DescriptionEntity,
+	id core.Identifier) (*models.DescriptionEntity, error) {
+	ctx := context.Background()
+	if descriptionEntity == nil {
+		return nil, nil
+	}
+
+	var longDescriptionBytes []byte
+	var sourceCode models.SourceCode
+	var err error
+
+	if descriptionEntity.LongDescription != nil {
+		longDescriptionBytes, err = proto.Marshal(descriptionEntity.LongDescription)
+		if err != nil {
+			logger.Errorf(ctx, "Failed to marshal LongDescription with error: %v", err)
+			return nil, err
+		}
+	}
+
+	if descriptionEntity.LongDescription != nil {
+		longDescriptionBytes, err = proto.Marshal(descriptionEntity.LongDescription)
+		if err != nil {
+			logger.Errorf(ctx, "Failed to marshal LongDescription with error: %v", err)
+			return nil, err
+		}
+	}
+	if descriptionEntity.SourceCode != nil {
+		sourceCode = models.SourceCode{Link: descriptionEntity.SourceCode.Link}
+	}
+
+	return &models.DescriptionEntity{
+		DescriptionEntityKey: models.DescriptionEntityKey{
+			ResourceType: id.ResourceType,
+			Project:      id.Project,
+			Domain:       id.Domain,
+			Name:         id.Name,
+			Version:      id.Version,
+		},
+		ShortDescription: descriptionEntity.ShortDescription,
+		LongDescription:  longDescriptionBytes,
```

Comment:
in an earlier version of this code, the long description was in S3 or a blob store of some kind right?  it now will store everything in the db?

Reply:
What are the length restrictions on the long description?

Reply:
4KB, flytekit will offload the long description if it exceeds 4KB when registering